### PR TITLE
Iteration5

### PR DESCRIPTION
**Pull request by abounk**
Create pub/sub and Notification system.

## PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others. (Please describe)

### Changes

1. Return Error when there's duplicate reminder time.
2. Create Pub/Sub system.
3. Create notification system, to remind user through Line application.
image

### Fixes

1. BUG: Assignment due date cannot be None.
2. BUG: Cannot send notification if tag's subscriber is None.

### CODE REVIEW

File: garnbarn_api/services/notification/platforms/notification_platform.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        if assignment_obj.tag and assignment_obj.tag.subscriber:
+            tag_subscriber = assignment_obj.tag.subscriber
+            subscriber = self._convert_subscriber_to_obj(tag_subscriber)
+            receiver = receiver.union(subscriber)
+
+        for user in receiver:
+            try:
+                self.notification_handler(assignment_obj, user)
+            except Exception as e:
+                logger.warn(
```
Comment (sirateek) [inline garnbarn_api/services/notification/platforms/notification_platform.py:23]: Maybe Change to logger.error?

File: garnbarn_api/services/notification/send_notification.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from django.dispatch.dispatcher import receiver
+from garnbarn_api.services.notification.platforms.notification_platform import NotificationPlatform
+from garnbarn_api.services.notification.platforms.line import Line
```
Comment (sirateek) [inline garnbarn_api/services/notification/send_notification.py:3]: This line is unnecessary

Comment (abounk) [reply garnbarn_api/services/notification/send_notification.py:3]: .__subclasses__ won't work if we don't import the classes.

**Review by SoSorryTT — APPROVED** [review]
Look good to me.

**Review by sirateek — APPROVED** [review]
LGTM

**Pull request merged**